File: storm-java21/src/test/java/st/orm/template/model/Owner.java
package st.orm.template.model;

import org.jspecify.annotations.Nullable;
import st.orm.Entity;
import st.orm.PK;
import st.orm.Version;

public record Owner(
        @PK Integer id,
        String firstName,
        String lastName,
        Address address,
        @Nullable String telephone,
        @Version int version
) implements Person, Entity<Integer> {}

File: storm-kotlin-spring/src/test/kotlin/st/orm/spring/model/Owner.kt
package st.orm.spring.model

import st.orm.Entity
import st.orm.PK
import st.orm.Version

/**
 * Simple domain object representing an owner.
 *
 */
@JvmRecord
data class Owner(
    @PK val id: Int = 0,
    override val firstName: String,
    override val lastName: String,
    val address: Address,
    val telephone: String?,
    @Version val version: Int
) : Entity<Int>, Person
